### ISSUE TITLE
Support custom thread-pools

### ISSUE THREAD

**Issue by foal** [amaembo/streamex#2]
Inspired by http://blog.krecan.net/2014/03/18/how-to-specify-thread-pool-for-java-8-parallel-streams/

**Comment by amaembo** [comment]
The idea is interesting, but may require some efforts in order not to produce any overhead for the users who don't need this functionality. I will think about it.

**Comment by amaembo** [comment]
I've implemented it in separate branch named custom-fjp. Now public API of all the streams contains new method: parallel(fjp). The terminal operation of the returned stream will be executed inside the supplied pool. If sequential() is called after that, then the resulting stream forgets about custom pool and the resulting stream will be executed as usual.

Extensive testing is necessary until I will be able to merge this branch. Especially it's necessary to check whether performance overhead for normal scenarios is acceptable. For now you may checkout this branch and test by yourself. If you wish, you may submit a pull-request with additional test cases (put them into CustomPoolTest class).

**Comment by amaembo** [comment]
The compact implementation actually produced significant performance overhead even for sequential streams, so I replaced it with less compact version which does not affect the performance (the final library jar becomes 20Kb larger, around 60Kb now). Nevertheless I think it's acceptable change, thus I merged it to the master. Will appear in the next public release (probably 0.2.0) after additional testing and documentation.
